feat(config): renovateCompatibility (#25855)
renovatebot/renovate: docs/usage/configuration-options.md

The field supports multiple URLs but it is datasource-dependent on whether only the first is used or multiple.

## renovateCompatibility

> This capability exists so that Renovate maintainers can change default behavior without forcing all existing users to adopt the new behavior immediately.

Reviewer: ```suggestion
We (the Renovate maintainers) created the `renovateCompatibility` config so that we can change Renovate's default behavior, without forcing users to use the new behavior immediately.
```

Author: I think I'd prefer it to be less personal/more objective, happy to remove references to maintainers altogether

Reviewer: Let's remove the reference to maintainers, it's not that interesting for Renovate end-users. Users only care about what the feature does for them. :wink:

Reviewer: ```suggestion
```

> Although Renovate adheres to SemVer and limits "breaking" behavior changes to major releases, the majority of _users_ of Renovate are end users in repositories and they do not have control over which major release of Renovate is running or when that major upgrade is performed.

Reviewer: ```suggestion
Renovate follows SemVer and limits "breaking" behavior changes to major releases.
This works well for users who self-host Renovate, because they control when Renovate is updated.
But most _users_ of Renovate are end users in repositories, they can't control which major release of Renovate is running or when Renovate updates.
```

> This is true particularly in the Renovate GitHub App for github.com, which is operated by the Renovate team at Mend.io.

Reviewer: ```suggestion
This is true particularly in the Mend Renovate app, which is operated by the Renovate team at Mend.io.
```

By introducing the `renovateCompatibility` feature it allows Renovate to maintain previous behavior whenever possible for repositories which were onboarded prior to the change in behavior.

> Users can then migrate their config to the new major release and decide that that point whether to adopt the new behavior, or keep a backwards compatibility setting to maintain the earlier behavior indefinitely.

Reviewer: ```suggestion
Users can then migrate their config to the new major release and decide at that point whether to adopt the new behavior, or keep a backwards compatibility setting to maintain the earlier behavior indefinitely.
```

> This setting was introduced in Renovate v38, so it has a default value of 37, meaning that if no compatibility value is defined then it assumes compatibility with v37.

Reviewer: ```suggestion
This setting was introduced in Renovate v38, so it has a default value of `37`, meaning that if no compatibility value is defined then it assumes compatibility with `v37`.
```

> This field can be set to either:

Reviewer: ```suggestion
You can set `renovateCompatibility` to:
```

> - 0, indicating "Do not care about compatibility, always give me the latest default behavior" (essentially the way most software works, including Renovate < v38), or

Reviewer: ```suggestion
- `0`, indicating "Do not care about compatibility, always give me the latest default behavior" (essentially the way most software works, including Renovate `< v38`), or
```

> - Any positive integer corresponding to a major release of Renovate (e.g. 37 or greater)

Reviewer: ```suggestion
- A positive integer that matches a major release of Renovate (for example `37`)
```
